### PR TITLE
docs: Use next-mdx-remote for rendering MDX content

### PR DESCRIPTION
**Pull request by roaks3**
Switches the docs site to use `next-mdx-remote` instead of `next-mdx-enhanced` to render MDX content. Changes that were required for this move:
- Moves MDX content from `pages` directory to the `content` directory (`docs`, `api-docs`, and `intro`)
- Moves MDX partials from `pages/partials` to `content/partials`

As a result of this change, the builds will also use significantly less memory and run faster.

---

Additional notes:
- Contributors to the docs site may need to `rm- rf website/.mdx-data` to remove artifacts previously created by `next-mdx-enhanced`

### CODE REVIEW

File: website/lib/docs-page.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      params: {
+        slug: mdxPath
+          .replace(/\.mdx$/, '')
+          .replace(/(^|\/)index$/, '')
+          .split('/'),
+      },
+    })),
+    fallback: false,
+  }
+}
```
Comment (jescalan) [inline website/lib/docs-page.js:80]: This seems to be almost exactly the same as the other implementations, only really different in terms of names of things and how the code is organized. You mentioned that there were some differences that cater to versioned docs, am I missing them here? Would be great if you could highlight them if so just so I'm aware. If there aren't any, it would probably be best to fully align the approach across sites where we have it implemented since the abstraction of this pattern is imminent.

**Review by jescalan — COMMENTED** [review]
This looks great, really just one question!

File: website/lib/docs-page.js (hunk 1 above)
Comment (roaks3) [reply website/lib/docs-page.js:80]: Sure, I think the easiest way to see all of the differences for Versioned Docs is by comparing this with the [Versioned Doc prototype](https://github.com/roaks3/versioned-doc-poc/blob/master/pages/docs/%5B...slug%5D.jsx). Here are some of the highlights:
- The version on the url will need to be extracted, and the version will need to be omitted when looking up the content
- The content will come from sources besides the filesystem for older versions, so it's most consistent to use promises, and we will not be able to rely on filesystem-specific functionality like `exists`
- The sidenav data will also come from sources besides the filesystem for older versions, so the implementation needs to support an alternative to `readAllFrontMatter()`, which will ideally be done in parallel with the content fetching
- Partials again have the same issue as the content and sidenav, and will need to consider the version being rendered

In short, these differences will require the implementation to be organized in a different way, and using different abstractions than when we only consider the single-version case, so I've largely kept things flat and flexible to those changes. Let me know if there is anything specific here you had concerns about.

**Comment by DingoEatingFuzz** [comment]
@jescalan I'll let you do the honors of closing this. I'm assuming your PR supersedes this one.
